asmcomp/cmm_helpers.ml

```diff
@@ -2188,18 +2200,19 @@ let assignment_kind
     (init: Lambda.initialization_or_assignment) =
   match init, ptr with
   | Assignment, Pointer -> Caml_modify
-  | Heap_initialization, Pointer -> Caml_initialize
+  | Heap_initialization, Pointer
+  | Root_initialization, Pointer -> Caml_initialize
```

Comment:
@ctk21 Can you add a comment about this change please? This is from https://github.com/ocaml-multicore/ocaml-multicore/commit/eda7c222972a049b7e18645a0f5e83553e044037#diff-9fdb20b499d1e0d702a1065cd7ec13849ff7c0ab944c25d52bcebcf2d3fc5983L1659-L1662.

Reply:
Related discussion: https://github.com/ocaml/ocaml/pull/10831/#discussion_r777185977.